lib/common/socket/evloop/epoll.c.h

```diff
@@ -82,6 +82,12 @@ static int update_status(struct st_h2o_evloop_epoll_t *loop)
                 } else {
                     assert(ev.events != 0);
                     op = EPOLL_CTL_ADD;
+                    if ((sock->_flags & H2O_SOCKET_FLAG_POLL_EXCLUSIVE) != 0) {
+#if defined(EPOLLEXCLUSIVE)
+                        /* EPOLLEXCLUSIVE: since Linux 4.5 glibc 2.24. */
+                        ev.events |= EPOLLEXCLUSIVE;
+#endif
+                    }
```

Comment:
I am not sure if this is the correct location to set `ev.events`, because it is specific when we do `EPOLL_CTL_ADD`.

Rather, I'd suggest doing that in line 53 by changing the code to `ev.events |= EPOLLIN | EPOLLEXCLUSIVE;`, assuming that `H2O_SOCKET_FLAG_POLL_EXCLUSIVE` changes the behavior of EPOLLIN (but not EPOLLOUT).

Reply:
@kazuho Added here because `EPOLLEXCLUSIVE` only works with `EPOLL_CTL_ADD`.

Reply:
Thank you for the clarification. That makes sense.